zrml/swaps/src/lib.rs

```diff
@@ -1444,9 +1460,13 @@ mod pallet {
                 .saturated_into();
                 let fee_pct = T::RikiddoSigmoidFeeMarketEma::fee(*pool_id)?.saturated_into();
                 let fee_plus_one = BASE.saturating_add(fee_pct);
-                let price_with_fee: u128 =
-                    bmul(fee_plus_one, bmul(price_with_inverse_fee, fee_plus_one)?)?;
-                Ok(price_with_fee.saturated_into())
+                let price_without_fee: u128 = bmul(price_with_inverse_fee, fee_plus_one)?;
```

Comment:
Ok, I understand now. Because inside the Rikiddo price we already have fees included, we need to multiply by `fee_plus_one` to get the fees out of the term.

```text
price_with_inverse_fee = 1 ZTG / (price + (1 + fee))
price_without_fee = (1 ZTG * (1 + fee)) / (price + (1 + fee)) = 1 ZTG / price
```